src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions_Tracking.cs

```diff
@@ -191,13 +192,13 @@ private static bool IsDescendant(SyntaxNode root, SyntaxNode node)
 
         private class CurrentNodes
         {
-            private readonly Dictionary<SyntaxAnnotation, IReadOnlyList<SyntaxNode>> _idToNodeMap;
+            private readonly ImmutableSegmentedDictionary<SyntaxAnnotation, IReadOnlyList<SyntaxNode>> _idToNodeMap;
```

Comment:
note: i do think we should doc these things.  perhaps with the AB link showing that this could go into the LOH.

Reply:
➡️ Link added

Reply:
@sharwell are there any benchmarks for this type? Unlike netfw/netcore, mono has a different rule for putting objects on the LOH.

I just wanted to take a look at some numbers for it, so I can write benchmarks myself if needed.

Reply:
Some are here https://github.com/dotnet/roslyn/blob/main/src/Tools/IdeCoreBenchmarks/SegmentedArrayBenchmarks_Indexer.cs